mlrun/api/crud/marketplace.py

```diff
@@ -183,22 +194,48 @@ def get_source_catalog(
     def get_item(
         self,
         source: MarketplaceSource,
-        item_name,
-        channel,
-        version=None,
-        tag=None,
-        force_refresh=False,
+        item_name: str,
+        version: Optional[str] = None,
+        tag: Optional[str] = None,
+        force_refresh: bool = False,
+    ) -> MarketplaceItem:
+        """
+        Retrieve item from source. The item is filtered by tag and version.
+
+        :param source:          Marketplace source object
+        :param item_name:       name of the item to retrieve
+        :param version:         version of the item
+        :param tag:             tag of the item
+        :param force_refresh:   if True, the catalog will be loaded from source always,
+                                otherwise will be pulled from db (if loaded before)
+        :return: marketplace item object
+        """
+        catalog = self.get_source_catalog(source, version, tag, force_refresh)
+        return self._get_item_from_catalog(catalog.catalog, item_name, version, tag)
+
+    @staticmethod
+    def _get_item_from_catalog(
+        catalog: List[MarketplaceItem], item_name, version, tag
```

Comment:
There's no longer any point in passing the `version` and `tag` parameters here - you're just using them for logging errors, and the `catalog` itself is already assumed to be filtered by them.
Probably best to remove them, and return how many items are there, and do the `>1` check in the calling function with an assertion there. If you do this, should probably also call the function differently, something like `_get_catalog_items_filtered_by_name`